File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string str) {
        stack<char> s;
        int flag =0;
        for (char c: str){
            if (c=='(' || c== '{' || c=='['){
                s.push(c);
            }
            else{
                if(s.empty()){
                    return false;
                    break;
                }
                char ch = s.top();
                s.pop();
                if((c==')' && ch=='(') || (c=='}' && ch=='{')|| (c==']' && ch== '[')){
                    continue;
                }
                else{
                    return false;
                    break;
                }
            }
        }
        return s.empty();
        
    }    
    
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        for c in s:
            if c in '({[':
                st.append(c)
            else:
                if len(st)==0:
                    return False
                    break
                    
                top = st[-1]
                st.pop()
                if (c == ')' and top != '(') or (c=='}' and top != '{') or (c==']' and top!= '['):
                    return False
                    break
                    

        return len(st) == 0
        